### PR TITLE
Remove some lesser-used reexports from Blammo.Logging

### PR DESCRIPTION
**Pull request by chris-martin**
Some of the changes mentioned in https://github.com/freckle/megarepo/pull/35369#issuecomment-2249189218 - A few obvious ways to trim down the main module with a focus on what most modules need to emit log messages.

### CODE REVIEW

File: Blammo/README.lhs [hunk 1]
```diff
@@ -44,6 +44,8 @@ import Control.Monad.Reader (MonadReader, ReaderT (runReaderT))
 
 ```haskell
 import Blammo.Logging.Simple
+import Blammo.Logging.LogSettings
+import Blammo.Logging.ThreadContext
```
Comment (pbrisbin) [inline Blammo/README.lhs:48]: I dunno, I might consider moving some of the re-exporting to `.Simple`, so this doesn't change. I thought of the name "simple" here as "all you need" or "batteries included", not as in "minimal". Seeing additional imports needed in the documentation tells me it's no longer completing the brief, so to speak.

File: Blammo/src/Blammo/Logging/ThreadContext.hs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module Blammo.Logging.ThreadContext
+  ( MonadMask
+  , withThreadContext
+  , myThreadContext
+  , Pair
```
Comment (pbrisbin) [inline Blammo/src/Blammo/Logging/ThreadContext.hs:5]: It's a bit irksome that the "pairs" in `Message` are `SeriesElem`, but the "pairs" in thread context are `Pair`.

**Review by pbrisbin — APPROVED** [review]

File: Blammo/CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,17 @@
-## [_Unreleased_](https://github.com/freckle/blammo/compare/Blammo-v2.0.0.0...main)
+## [_Unreleased_](https://github.com/freckle/blammo/compare/Blammo-v2.2.0.0...main)
+
+## [v2.1.0.0](https://github.com/freckle/blammo/compare/v2.0.0.0...Blammo-v2.1.0.0)
```
[the remark this answers is not in the document]
Comment (chris-martin) [reply Blammo/CHANGELOG.md:3]: I guess so. I like to lean toward the [more releases](https://www.haskellforall.com/2019/05/release-early-and-often.html) philosophy, but not strongly.

[the remark this answers is not in the document]
Comment (pbrisbin) [reply Blammo/CHANGELOG.md:3]: Yeah, that post is compelling, and I guess I could be wrong that multiple major version bumps even comes with downsides. In my head, I was considering Stackage LTS, and how a major bump means that version won't appear until the next major LTS. So I have this intuition that minimizing major bumps would maximize how often we will get the latest of our own packages without resorting to `extra-deps`. But there's a lot of moving parts and timing to consider to know if batching it or not will actually work out to any tangible difference in practice. And `extra-deps` isn't even a major drawback, so :shrug:

**Pull request merged**